perfkitbenchmarker/digitalocean/digitalocean_virtual_machine.py

```diff
@@ -0,0 +1,194 @@
+      time.sleep(wait_seconds)
+      stdout, ret = util.RunCurlCommand('GET', 'actions/%s' % action_id)
+      if ret != 0:
+        logging.warn('Unexpected action lookup failure: %s', stdout)
+        return False
+      response = json.loads(stdout)
+      status = response['action']['status']
+      logging.debug('action %d: status is "%s".', action_id, status)
+      if status == 'completed':
+        return True
+      elif status == 'errored':
+        return False
+    # If we get here, waiting timed out. Treat as failure.
+    logging.debug('action %d: timed out waiting.', action_id)
+    return False
+
+  def _PostCreate(self):
+    """Get the instance's data."""
+    stdout, _ = util.RunCurlCommand(
+        'GET', 'droplets/%s' % self.droplet_id)
+    response = json.loads(stdout)['droplet']
+    network_interface = response['networks']['v4'][0]
+    # Not currently using private addresses. Do we need to?
+    self.internal_ip = network_interface['ip_address']
+    self.ip_address = network_interface['ip_address']
+
+  def _Delete(self):
+    """Delete a DigitalOcean VM instance."""
+    stdout, ret = util.RunCurlCommand(
+        'DELETE', 'droplets/%s' % self.droplet_id)
+    if ret != 0:
+      raise errors.Resource.RetryableDeletionError(
+          'Deletion failed: %s' % stdout)
+
+    # Get the droplet's actions so that we can look up the
+    # ID for the deletion just issued.
+    stdout, ret = util.RunCurlCommand(
+        'GET', 'droplets/%s/actions' % self.droplet_id)
+    if ret != 0:
+      # There's a race condition here - if the lookup fails, assume it's
+      # due to deletion already being complete. Don't raise an error in
+      # that case,  the _Exists check should trigger retry if needed.
+      return
+    response = json.loads(stdout)['actions']
+
+    # Get the action ID for the 'destroy' action. This assumes there's only
+    # one of them, but AFAIK there can't be more since 'destroy' locks the VM.
+    destroy = [v for v in response if v['type'] == 'destroy'][0]
+
+    # Wait for completion. Actions are global objects, so the action
+    # status should still be retrievable after the VM got destroyed.
+    # We don't care about the result, let the _Exists check decide if we
+    # need to try again.
+    self._GetActionResult(destroy['id'])
+
+  def _Exists(self):
+    """Returns true if the VM exists."""
+    stdout, ret = util.RunCurlCommand(
+        'GET', 'droplets/%s' % self.droplet_id,
+        suppress_warning=True)
+    return ret == 0
+
+  def GetScratchDir(self, disk_num=0):
+    """Gets the path to the scratch directory.
+
+    DigitalOcean does not support separate disk devices at this time.
+    Use a directory on the main disk as a single scratch dir, but fail
+    if a test assumes multiple distinct disks, i.e. for copy benchmark.
+
+    Args:
+      disk_num: The number of the disk to mount.
+    Returns:
+      The mounted disk directory.
+    """
+
+    if disk_num > 0:
```

Comment:
Maybe just add this check (assert disk_num is a valid disk) to virtual_machine.BaseVirtualMachine.GetScratchDir() and delete this method